packages/medusa/src/api-v2/admin/products/helpers.ts

```diff
@@ -0,0 +1,46 @@
+import { ProductDTO, ProductVariantDTO } from "@medusajs/types"
+
+// The variant had prices before, but that is not part of the price_set money amounts. Do we remap the request and response or not?
+export const remapKeysForProduct = (selectFields: string[]) => {
+  const productFields = selectFields.filter(
+    (fieldName: string) => !fieldName.startsWith("variants.prices")
+  )
+  const pricingFields = selectFields
+    .filter((fieldName: string) => fieldName.startsWith("variants.prices"))
+    .map((fieldName: string) =>
+      fieldName.replace("variants.prices.", "variants.price_set.money_amounts.")
```

Comment:
suggestion: once we do the model changes in pricing, we should just return the exact shape we would get from the remote query, price_set included. I don't see any reason to hide it from the consumers.

Reply:
@riqwan we talked with Oli that we want to keep the model simple, as it is more intuitive to return just a prices entry. This is the whole point of decoupling the HTTP from module endpoints, so we can map to whatever works best for the API in mind.